### ISSUE TITLE
Make lazy load errors easily identifiable

### ISSUE THREAD

**Issue by Ephem** [gregberge/loadable-components#717]
## 🚀 Feature Proposal

Add some way to identify errors from lazy loading.

## Motivation

Because error handling in loadable happens in error boundaries, it's useful to be able to discern between lazy load errors and other errors so you can handle them differently. Today there is no easy way (that I know of) to know if an error that has bubbled to an error boundary comes from lazy loading a component, or something else.

## Example

Maybe something like this? It would be nice if this also worked for things like timeouts that use an external library, that is, decorate any error that happens on lazy load.

```jsx
import { isLoadableError } from '@loadable/component';

class ErrorBoundary extends React.Component {
  componentDidCatch(error) {
    if (isLoadableError(error)) {
      // TODO
    } else {
      // TODO
    }
  }
  render() { /* ... */ }
 }
```

## Pitch

Today you could achieve this in userland by building a comprehensive list of any error that could happen and match against that, but this is cumbersome, error prone and isn't safe against any future errors that could get introduced into the library etc.

**Comment by Ephem** [comment]
Bump, this is still an active feature proposal.

**Comment by theKashey** [comment]
That would be nice to align this one with `React.lazy` if not with the origin of error(bundler)

But in general I do approve this idea

**Comment by Ephem** [comment]
I agree! I've been scrolling the new [React 18 working group](https://github.com/reactwg/react-18/discussions) but haven't seen much details around `React.lazy` just yet.

I'd love if [this CodeSandbox example](https://codesandbox.io/s/github/facebook/react/tree/master/fixtures/ssr2?file=/src/App.js) showed how to render different errors in the `<Error>`-component based on if it was `React.lazy` that failed, but I mean to play around with it and log some errors to see if I can get some good leads. 😄

**Comment by theKashey** [comment]
Side note: can help with https://github.com/gregberge/loadable-components/issues/716
